config.py

```diff
@@ -8,7 +8,7 @@
 
 dir_path = os.path.dirname(os.path.realpath(__file__))
 
-spack_version = 'v0.15.4'
+spack_version = 'v0.17.0'
```

Comment:
I think this is a big enough change, and we should throughly test this independent of the rest of this PR. 
I would create a PR just for the update of v0.17.0, and wait until the test functionality of @PanicSheep is ready to test it

Reply:
this is the one then Carlos! I'll wait for the big test then, seems quite ok already though!

Reply:
I meant there are more changes here in addition to the version upgrade (i.e. compilers, pgi, individual packages etc). 
I think we should tests solely the effect of upgrading spack. Also we would need to test all other packages (not only cosmo) and all other usages documented in the documentation, i.e. load, dev-build, devbuildcosmo, etc

Reply:
Test functionality is ready.